<a href="https://colab.research.google.com/github/sudha-vista/InsightTracker/blob/main/Day33_Amstrong_number.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:


n = int(input("Enter a number: "))

#Keep a copy to compare later
original = n

sum_of_cubes = 0

#Repeat until number becomes 0
while n > 0:
    digit = n % 10          # Get the last digit
    sum_of_cubes += digit ** 3   # Add cube of digit to sum
    n = n // 10             # Remove the last digit

#Compare sum with original
if sum_of_cubes == original:
    print(original, "is an Armstrong number ")
else:
    print(original, "is NOT an Armstrong number ")

Enter a number: 407
407 is an Armstrong number 


* Short definition:

  A number is Armstrong if sum of cubes of its digits = the number itself.




* Modulus (%) → normally gives remainder

But when you do % 10, remainder = last digit (because 10 fits all digits except the last one)


* // → Floor Division (Whole number division)

It removes the last digit

👉 Example:

153 // 10   # means: divide 153 by 10, but keep only the whole number

🧮 153 ÷ 10 = 15.3 → only 15 is kept ✅

That’s how we cut off the last digit.


* Each round → one digit is removed.

After 3 rounds → no digits left → number becomes 0.

Loop ends!


* Check if 407 is an Armstrong number

Step 1: Write digits → 4, 0, 7
Step 2: Cube each digit →

4³ = 64

0³ = 0

7³ = 343


Step 3: Add them →
64 + 0 + 343 = 407